osf/migrations/0137_add_fm_record_to_osfstorage_files.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-09-26 15:18
+from __future__ import unicode_literals
+
+from django.db import migrations, connection
+
+
+def remove_records_from_files(state, schema):
+    FileMetadatRecord = state.get_model('osf', 'filemetadatarecord')
+    FileMetadatRecord.objects.all().delete()
+
+def add_records_to_files_sql(state, schema):
+    FileMetadataSchema = state.get_model('osf', 'filemetadataschema')
+    datacite_schema_id = FileMetadataSchema.objects.filter(_id='datacite').values_list('id', flat=True)[0]
+    sql = """
+    INSERT INTO osf_filemetadatarecord (created, modified, _id, metadata, id, file_id, schema_id)
+    SELECT  NOW(), NOW(), generate_object_id(), '{}', nextval('osf_filemetadatarecord_id_seq'), OSF_FILE.id, %d
+        FROM osf_basefilenode OSF_FILE
+            LEFT OUTER JOIN osf_filemetadatarecord ON
+                (OSF_FILE.id = osf_filemetadatarecord.file_id)
+            WHERE (OSF_FILE.type = 'osf.osfstoragefile'
+                   AND OSF_FILE.provider = 'osfstorage'
+                   AND osf_filemetadatarecord.id IS NULL
+            );
+    """ % (datacite_schema_id)
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0136_add_datacite_file_metadata_schema'),
+    ]
+
+    operations = [
+        migrations.RunPython(add_records_to_files_sql, remove_records_from_files),
```

Comment:
I didn't test this on a large data dump, but if it's slow, we might try RunSQL instead.

Reply:
This migration used to be 0134, when that was the case here were the benchmarks [from an earlier comment](https://github.com/CenterForOpenScience/osf.io/pull/8721#discussion_r221661404) -- On about 1.3 million files locally the migration ran for 1 min 50 seconds.

Reply:
wow, okay!

Reply:
Matt says the migration took 30 min on prod data which is pushing it. Looking into optimizing this.